visualisation/single_spectra.py

```diff
@@ -0,0 +1,110 @@
+import pandas as pd
+import peakutils
+import plotly.graph_objects as go
+import streamlit as st
+
+from processing import save_read
+from processing import utils
+from . import draw
+
+SINGLE = 'Single spectra'
+MS = "Mean spectrum"
+GS = "Grouped spectra"
+P3D = "Plot 3D"
+
+AV = "Average"
+BS = "Baseline"
+RS = "Raman Shift"
+DS = "Dark Subtracted #1"
+DEG = "Polynominal degree"
+WINDOW = "Set window for spectra flattening"
+DFS = {'ML model grouped spectra': f'{DS}', 'ML model mean spectra': f'{AV}'}
+FLAT = "Flattened"
+COR = "Corrected"
+ORG = "Original spectrum"
+RAW = "Raw Data"
+OPT = "Optimised Data"
+NORM = "Normalized"
+OPT_S = "Optimised Spectrum"
+
+
+def show_single_plots(df, plots_color, template, spectra_conversion_type):
+    global plot_line, description, fig_single_all, corrected_df, df_visual, col2, col1
+    df2 = df.copy()
+    
+    for col in range(len(df2.columns)):
```

Comment:
Wolałbym mieć funkcję, która robi coś takiego dla jednej kolumny.
I wywołać ją w pętli. Jakbyśmy tę pętlę wstawili do `vis.py` to nawet zamiast "===============" można by da nagłówek z nazwą kolumny, albo coś...

wtedy nawet nie potrzebowałbyś robić kopii `df`. 

Tu też rozdzielamy przetwarzanie danych (zwracamy series), robienie plota, rysowanie plota i zapisywanie wyników przetwarzania danych.